piksi_tools/interpolate_event_positions.py

```diff
@@ -0,0 +1,246 @@
+      leftbound = msg_tow[itt]
+    itt+=1
+  return leftbound
+
+def get_rightbound (message_type, msg_tow, trigger_tow,msgout, numofmsg):
+  itt=0
+  rightbound=msg_tow[numofmsg-1]
+  while itt<numofmsg:
+    if msg_tow[itt]>trigger_tow and msg_tow[itt]<rightbound and message_type[itt] == msgout:
+      rightbound = msg_tow[itt]
+    itt+=1
+  return rightbound
+def get_position_parameter(message_type,msg_tow,msg_position, tow, numofmsg, msgout):
+  itt=0
+  value=0
+  while itt<numofmsg:
+    if msg_tow[itt]==tow and message_type[itt]== msgout:
+      value= msg_position[itt]
+    itt+=1
+  return value
+
+def get_trigger_positions(message_type,msg_tow,msgout,numofmsg, msg_horizontal, msg_vertical, msg_depth):
+  # extracts the position at the trigger tow using interpolation subroutine
+  itt=0
+  while itt<numofmsg:
+    if message_type[itt] == "MsgExtEvent" and msg_tow[itt] > 0:
+      #get left and right tow bounds 
+      left = get_leftbound(message_type,msg_tow,msg_tow[itt],msgout,numofmsg)
+      right = get_rightbound(message_type,msg_tow,msg_tow[itt],msgout,numofmsg)
+
+      # get left and right vertical horizontal and depth positions
+      left_h = get_position_parameter(message_type,msg_tow,msg_horizontal, left, numofmsg, msgout)
+      right_h = get_position_parameter(message_type,msg_tow,msg_horizontal, right, numofmsg, msgout)
+      left_v = get_position_parameter(message_type,msg_tow,msg_vertical, left, numofmsg, msgout)
+      right_v= get_position_parameter(message_type,msg_tow,msg_vertical, right, numofmsg, msgout)
+      left_d= get_position_parameter(message_type,msg_tow,msg_depth, left, numofmsg, msgout)
+      right_d= get_position_parameter(message_type,msg_tow,msg_depth, right, numofmsg, msgout)
+
+
+      # interpolate trigger position
+      msg_horizontal[itt]=lin_interp(left_h, right_h, left, right, msg_tow[itt])
+      msg_vertical[itt]=lin_interp(left_v, right_v, left, right, msg_tow[itt])
+      msg_depth[itt]=lin_interp(left_d, right_d, left, right, msg_tow[itt])
+    itt+=1
+  return 
+
+def display_data(message_type,msg_tow,msg_horizontal,msg_vertical,msg_depth,msgtype,outfile, numofmsg,msg_flag):
+  fout= open(outfile,'wt')
+  writer = csv.writer(fout)
+  if msgtype == 'MsgBaselineNED' :
+    indexdata = ("TOW (ms)", "N (mm)", "E (mm)", "D (mm)")
+  elif msgtype == 'MsgPosECEF' or msgtype == 'MsgBaselineECEF' :
+    indexdata = ("TOW (ms)", "X (m)", "Y (m)", "Z (m)")
+  elif msgtype == 'MsgPosLLH' :
+    indexdata = ("TOW (ms)", "Lat (deg)", "Lon (deg)", "Height (m)")
+  writer.writerow(indexdata + ("# of Sats", "Flags"))
+
+  itt=0
+  while itt<numofmsg:
+    if message_type[itt]=="MsgExtEvent" and msg_tow[itt] > 0 :
+      writer.writerow((msg_tow[itt],msg_horizontal[itt],msg_vertical[itt],msg_depth[itt],"0",msg_flag[itt]))
+    itt+=1
+  return 
+
+
+
+
+
+def write_positions(infile, outfile, msgtype, debouncetime):
```

Comment:
Change this to "collect_positions" a function that gets all the trigger positions and puts them as a list of interpolated dictionaries for each message. For example:  if there were 2 msg_ext_events, and you wanted the baseline message, you'd get [{tow: 10000, n: 1, e:2, d:3}, {tow: 10010, n: 1, e:2, d:3}].  This way we can use the intermediate output programatically.

Then we'll make a separate library function that writes a list of dictionaries to CSV, which is basically built into the CSV module.